.github/workflows/build.yml

```diff
@@ -0,0 +1,195 @@
+    runs-on: ubuntu-latest
+    container: ${{matrix.container}}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        in-process: [all, none]
+        container: ['debian:testing', 'ubuntu:rolling', 'archlinux:latest']
+
+    env:
+      CC: ccache gcc
+      CXX: ccache g++
+      BASE_INSTALL_PATH: ${{github.workspace}}/_install
+
+    steps:
+      - name: Setup environment
+        run: |
+          echo "PATH=${BASE_INSTALL_PATH}/bin:${PATH}" >> "$GITHUB_ENV"
+          echo "PKG_CONFIG_PATH=${BASE_INSTALL_PATH}/lib/pkgconfig:${PKG_CONFIG_PATH}" >> "$GITHUB_ENV"
+
+      - name: Show environment
+        run: env | sort
+        if: ${{ env.DEBUG == '1' }}
+
+      # For Debian and Ubuntu (apt-based with reasonably compatible packages)
+      - name: Install dependencies
+        if: ${{ startsWith(matrix.container, 'debian:') || startsWith(matrix.container, 'ubuntu:') }}
+        run: |
+          apt-get update -qq
+          apt-get install --assume-yes --no-install-recommends \
+            ${DEB_BUILD_DEPS} ${DEB_LIBRARY_DEPS}
+
+      # For ArchLinux
+      - name: Install dependencies
+        if: ${{ startsWith(matrix.container, 'archlinux:') }}
+        # don't upgrade, although told otherwise (see link below), because
+        # apparently in the container it doesn't quit work...
+        # https://wiki.archlinux.org/title/System_maintenance#Partial_upgrades_are_unsupported
+        run: |
+          pacman --noconfirm -Sy
+          pacman --noconfirm -S ${ARCH_BUILD_DEPS}
+
+      - uses: actions/checkout@v3
+        with:
+          path: mate-panel
+          submodules: true
+
+      # TODO: cache mate-desktop build/install?
+      - name: Checkout mate-desktop
+        uses: actions/checkout@v3
+        with:
+          repository: mate-desktop/mate-desktop
+          ref: v1.27.1
+          path: mate-desktop
+          submodules: true
+
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1.2
+
+      - name: Configure mate-desktop
+        run: |
+          cd mate-desktop
+          NOCONFIGURE=1 ./autogen.sh
+          { ./configure --prefix="$BASE_INSTALL_PATH" || { cat config.log; exit 1; } ; }
+
+      - name: Build mate-desktop
+        run: make -C mate-desktop -j ${{ env.JOBS }}
+
+      - name: Install mate-desktop
+        run: make -C mate-desktop -j ${{ env.JOBS }} install
+
+      - name: Configure
+        run: |
+          cd mate-panel
+          NOCONFIGURE=1 ./autogen.sh
+          { ./configure ${CONFIGURE_FLAGS} --with-in-process-applets=${{matrix.in-process}} || { cat config.log; exit 1; } ; }
+
+      - name: Build
+        run: make -C mate-panel -j ${{ env.JOBS }}
+
+      - name: Run Tests
+        run: make -C mate-panel -j ${{ env.JOBS }} check
+
+      - name: Run distcheck
+        if: ${{ startsWith(matrix.container, 'debian:') }}
+        run: make -C mate-panel -j ${{ env.JOBS }} distcheck
```

Comment:
As mentioned in the commit comment, I chose to only do distcheck on on system, as I see a lot less value in doing this everywhere.  But we can enable it back for every one if need be.

Reply:
I think it's sufficent to run run distcheck only one time.

Reply:
well, twice (one with in-process and out-of-process), as it might affect this, but yeah.